### PR TITLE
107: CI to push automated gov.txt updates from CZDS API

### LINKED ISSUE #107: Update gov.txt daily from CZDS
This repo is a public version of some of .gov's data. The zone file is in CZDS (Centralized Zone Data Service), but we should house a daily copy here. Use https://github.com/icann/czds-api-client-python. 

Acceptance Criteria:

- [x] add an ADR on our approach
- [x] an automated process is created thru which the .gov text file is updated daily, in the dotgov-data repository, from CZDS.

### PR DESCRIPTION
**Pull request by erinysong**
Resolves #107 - Creates GitHub Action workflow that daily pulls .gov zone file from CZDS API and pushes zone file contents to gov.txt if a diff is found. Script also allows for manually running workflow once it is merged to main.

Workflow currently triggered on push to es/107-zone-file-script for testing and demo purposes. This configuration will be removed once the branch is merged to main and it is confirmed to successfully run on main.

### Testing Locally (Requires an ICANN account)
1. In the command line, run `./scripts/download_zone_file.sh <ICANN_email> <ICANN_password>`. If you need an account to test, message @erinysong and I can provide you testing credentials.
2. The script should download and unzip the .gov zone file, resulting in an updated gov.txt

### Testing on GitHub Actions
Until this PR is pushed to main, this workflow cannot be run manually and for testing purposes is run on any push or PR to this branch (this is admittedly not the most ideal way to test and will be deleted later to do more thorough testing with the manual workflow option after merging). Make any temporary push to the branch and check for GitHub Actions to run the **Update gov.txt from CZDS API zone file** action. The action should successfully run.

### CODE REVIEW

File: .github/workflows/update-gov-zone-file.yml
Comment (h-m-f-t) [inline .github/workflows/update-gov-zone-file.yml]: Can we call our user `botgov`? 😄

File: .github/workflows/update-gov-zone-file.yml
Comment (h-m-f-t) [inline .github/workflows/update-gov-zone-file.yml]: Should we tie this to a non-user account (maybe make one for help@get.gov)?

File: .github/workflows/update-gov-zone-file.yml
Comment (h-m-f-t) [inline .github/workflows/update-gov-zone-file.yml]:
```suggestion
          git commit -m "CZDS update ${timestamp}" || exit 0
```

File: current-full.csv
Comment (h-m-f-t) [inline current-full.csv:1]: Can you say a bit about why this PR needs to touch current-full?

**Review by h-m-f-t — COMMENTED** [review]
Nice work @erinysong! I'll leave the real review to other engineers, but wanted to share a few comments.

File: current-full.csv
Comment (erinysong) [reply current-full.csv:1]: It shouldn't and just fixed it (accidentally deleted this file when trying to remove another one and restored it with an outdated version of the file) - thank you for noting that!

File: .github/workflows/update-gov-zone-file.yml
Comment (erinysong) [reply .github/workflows/update-gov-zone-file.yml]: I like that a lot and laughed - will change it :D

File: .github/workflows/update-gov-zone-file.yml
Comment (erinysong) [reply .github/workflows/update-gov-zone-file.yml]: I'd be open to that! As a disclaimer I don't know too much about having GitHub accounts specifically for orgs to have an opinion on this

File: .github/workflows/update-gov-zone-file.yml
Comment (CocoByte) [inline .github/workflows/update-gov-zone-file.yml]: I assume this will be cleaned up before merging...?

File: .github/workflows/update-gov-zone-file.yml
Comment (CocoByte) [reply .github/workflows/update-gov-zone-file.yml]: I second what Cameron is saying. ..Unless you want to be forever tied to this action, Erin?  :D

**Review by CocoByte — COMMENTED** [review]
Overall looks good, just a couple comments alongside Cameron's

File: .github/workflows/update-gov-zone-file.yml
Comment (erinysong) [reply .github/workflows/update-gov-zone-file.yml]: Yes! will remove this once this is approved - after merging this workflow should be accessible to run manually which will make testing easier

**Review by therealslimhsiehdy — APPROVED** [review]
Had a Zoom call with Erin and saw her demo it live to show that it worked + asked some clarification questions. Approving with the caveat that Erin will remove the lines that are for local testing! YAY ERIN!!!

**Comment by erinysong** [comment]
Merging PR and will continue with testing on main branch to assess workflow health and functionality.

**Pull request merged**
